sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -131,6 +131,26 @@ case class CreateViewCommand(
         s"specified by CREATE VIEW (num: `${userSpecifiedColumns.length}`).")
     }
 
+    // When creating a permanent view, not allowed to reference temporary objects.
+    if (!isTemporary) {
+      child.collect {
+        // Disallow creating permanent views based on temporary views.
```

Comment:
i'd also copy paste what you put in the pr description in here, on why you are traversing the unresolved plan.